### PR TITLE
use unprojectPoint to find transform

### PR DESCRIPTION
**Pull request by XanderXu**

### CODE REVIEW

File: ARKit-QRScanner/QRScanner.swift [hunk 1]
```diff
@@ -11,59 +11,303 @@ import ARKit
...
+            let topRightInView = cgpointsInView[2]
+            let topLeftInView = cgpointsInView[3]
+            let centerInView = cgpointsInView[8]
+            
+            var bottomLeftRough = simd_float3(repeating: 0)
+            var bottomRightRough = simd_float3(repeating: 0)
+            var topRightRough = simd_float3(repeating: 0)
+            var topLeftRough = simd_float3(repeating: 0)
+            
+            let points = cgpointsInImage.reduce(into: [simd_float3]()) { (points, cgpoint) in
```
Comment (maxxfrazer) [inline ARKit-QRScanner/QRScanner.swift:57]: Looks like a `compactMap` would work better for you here
https://developer.apple.com/documentation/swift/sequence/2950916-compactmap

File: ARKit-QRScanner/QRScanner.swift [hunk 2]
```diff
@@ -11,59 +11,303 @@ import ARKit
...
+        let imageWidth = CGFloat(CVPixelBufferGetWidth(capturedImage))//1920
+        let imageHeight = CGFloat(CVPixelBufferGetHeight(capturedImage))//1440
+        
+        let factorH = viewSize.height / imageWidth
+        let factorW = viewSize.width / imageHeight
+        let factor = CGFloat.minimum(factorH, factorW)
+        
+        
+        
+        return features.reduce(into: [QRResponse]()) { (responses, feature) in
```
Comment (maxxfrazer) [inline ARKit-QRScanner/QRScanner.swift]: Looks like this is repeated code of the previous function. Could you take out this functionality and put it into a private method?

**Review by maxxfrazer — CHANGES_REQUESTED** [review]
Thank you @XanderXu for making a Pull Request! It's one of the first pull request I've received on a public repository so means a lot.

Would it be possible for you to remove the duplicated code and consolidate it into a private function? With the parameter being the features, response being `[QRResponse]`

Also there's now a mix of tabs/spaces throughout the code, but once you're done I'm going to change everything to 2 spaces, so don't worry about that.

Thanks again for the pull request!

File: ARKit-QRScanner/QRScanner.swift (hunk 2 above)
Comment (XanderXu) [reply ARKit-QRScanner/QRScanner.swift]: OK

**Pull request closed without merging**